File: backend/app/entities/session.go
package entities

type Session struct {
	Base
	UserID    uint      `json:"user_id" gorm:"not null"`
	ChatbotID uint      `json:"chatbot_id" gorm:"not null"`
	User      User      `json:"user,omitempty" gorm:"foreignKey:UserID"`
	Chatbot   Chatbot   `json:"chatbot,omitempty" gorm:"foreignKey:ChatbotID"`
	Messages  []Message `json:"messages,omitempty" gorm:"foreignKey:SessionID"`
}

File: frontend/src/types/session.ts
import { IUser } from './user'
import { IChatbot } from './chatbot'

export interface IMessage {
  id: number
  session_id: number
  role: string
  content: string
  session?: ISession
  created_at?: string
  updated_at?: string
}

export interface ISession {
  id: number
  user_id: number
  chatbot_id: number
  user?: IUser
  chatbot?: IChatbot
  messages?: IMessage[]
  created_at?: string
  updated_at?: string
}
